pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -329,35 +359,124 @@ func (w *Webhook) getAllLatestLibraries() []libInfo {
 	return libsToInject
 }
 
-// extractLibInfo returns the language, the image,
-// and a boolean indicating whether the library should be injected into the pod
-func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
-	// If the pod is "injectable" and annotated with libraries to inject, use those.
-	if w.isPodEligible(pod) {
-		// The library version specified via annotation on the Pod takes precedence
-		// over libraries injected with Single Step Instrumentation
-		libs := w.extractLibrariesFromAnnotations(pod)
-		if len(libs) > 0 {
-			return libs, false
-		}
+// libInfoSource describes where we got the libraries from for
+// injection and is used to set up metrics/telemetry. See
+// Webhook.injectAutoInstruConfig for usage.
+type libInfoSource int
+
+const (
+	// libInfoSourceNone is no source provided.
+	libInfoSourceNone libInfoSource = iota
+	// libInfoSourceLibInjection is when the user sets up manual annotations on their pods.
+	libInfoSourceLibInjection
```

Comment:
nit:

I am not sure, but I think using `libInfoSourcePodAnnotations` is more conveying here. 

Without the comment above, I think it is not very intuitive to guess that this is used when the user sets up manual annotations on their pods.

Reply:
The reason I want to keep this as `libInjection` is that it's the name of the product (and how this ends up being logged in the environment variables and metrics).

It's not _exactly_ that it's just the annotations, but its the annotations with single step being disabled for the namespace-- updating the comment to that effect.